agent/test-agent-cli/tests/cli_test.rs

```diff
@@ -0,0 +1,76 @@
+#![cfg(feature = "integ")]
+mod data;
+use assert_cmd::Command;
+use model::{constants::NAMESPACE, Test};
+use selftest::Cluster;
+use tokio::time::Duration;
+
+const POD_TIMEOUT: Duration = Duration::from_secs(300);
+#[tokio::test]
+async fn test_system() {
+    let cluster_name = "integ-test";
+    let cluster = Cluster::new(cluster_name).unwrap();
+    cluster.load_image_to_cluster("controller:integ").unwrap();
+    let mut cmd = Command::cargo_bin("testsys").unwrap();
+    cmd.args(&[
+        "--kubeconfig",
+        cluster.kubeconfig().to_str().unwrap(),
+        "install",
+        "--controller-uri",
+        "controller:integ",
+    ]);
+    cmd.assert().success();
+    cluster.wait_for_controller(POD_TIMEOUT).await.unwrap();
+
+    cluster
+        .load_image_to_cluster("example-test-agent-cli:integ")
+        .unwrap();
+
+    let mut cmd = Command::cargo_bin("testsys").unwrap();
+    cmd.args(&[
+        "--kubeconfig",
+        cluster.kubeconfig().to_str().unwrap(),
+        "run",
+        "file",
+        data::integ_test_dependent_path().to_str().unwrap(),
+    ]);
+    cmd.assert().success();
+
+    cluster
+        .wait_for_test_pod("hello-world-cli", POD_TIMEOUT)
+        .await
+        .unwrap();
+
+    let mut cmd = Command::cargo_bin("testsys").unwrap();
+    cmd.args(&[
+        "--kubeconfig",
+        cluster.kubeconfig().to_str().unwrap(),
+        "status",
+        "-c",
+        "-r",
+        "--json",
+    ]);
+
+    tokio::time::sleep(Duration::from_secs(15)).await;
```

Comment:
Why do we need to sleep here?

Reply:
Else it collect results so quickly and status comes as running so I made it to wait for some time.

Reply:
Can you move the sleep above the command creation? It is a bit confusing to read.